### ISSUE TITLE
Problem loading local artwork file

### ISSUE THREAD

**Issue by ihutc** [react-native-kit/react-native-track-player#458]
I'm trying to use a locally stored image file (downloaded using RNFetchBlob) as the artwork for a track, but I'm having trouble getting it to appear in the notification centre controls.

Playing audio that's downloaded and stored locally works perfectly using an absolute path like `file:///var/mobile/Containers/Data/Application/BB86EFE9-1ED0-4AC8-9E95-A9B16EF23C97/Documents/22485.mp3`.

I've tried using a similar absolute path string for `artwork`, but it doesn't work. I've also tried wrapping the path in a resource object, like below
```
artwork: {
  uri: 'file:///var/mobile/Containers/Data/Application/BB86EFE9-1ED0-4AC8-9E95-A9B16EF23C97/Documents/22485.jpg'
}
```

Using an image stored online over http(s) works fine. I'm using iOS 12.1.3 on iPhone X.

**Comment by ihutc** [comment]
To get this working for now, I've made some _super hacky_ changes to the `setArtwork` function in `AudioPlayer.swift` in [this commit](https://github.com/ihutc/react-native-track-player/commit/a1e7b3d7b125a316174ab0d01bf5bb2be160273a).

It's pretty similar to what exists, but doesn't use `URLSession` to get the file. It instead uses an absolute URL of an image file in the file system, without the `file://` prefix and loads the image directly from that in a `Data` object.

I'm not super familiar with `URLSession` , but could it be a problem for local files?

**Comment by rkulinski** [comment]
Is there a hope that this issue will be fixed anytime soon?

**Comment by bishnudev** [comment]
has it been fixed?

**Comment by nishanBende** [comment]
@Guichaguri I updated getArtwork method in Track.swift to support local artworks. 
It works if I pass `file://` URIs in artwork property.
P.S. It can be done in a better way though. 

```

    func getArtwork(_ handler: @escaping (UIImage?) -> Void) {
        if let artworkURL = artworkURL?.value {
           
            // Handle Local image
            if(artworkURL.path.starts(with: "/")){
                let image = UIImage.init(contentsOfFile: artworkURL.path);
                handler(image);
            } else {
                
                URLSession.shared.dataTask(with: artworkURL, completionHandler: { (data, _, error) in
                    if let data = data, let artwork = UIImage(data: data), error == nil {
                        handler(artwork)
                    }
                    
                    handler(nil)
                }).resume()
            }
        }
        
        handler(nil)
    }
```

**Comment by curiousdustin** [comment]
@nishanBende's https://github.com/react-native-kit/react-native-track-player/pull/897 resolved this. Included in `v1.2.3`
